web/gds-user-ui/src/components/Modal/ConfirmationResetFormModal.tsx

```diff
@@ -12,73 +12,112 @@ import {
   useDisclosure,
   Button
 } from '@chakra-ui/react';
-import { getRegistrationDefaultValues } from 'modules/dashboard/certificate/lib';
-import { useNavigate } from 'react-router-dom';
+import { useDeleteCertificateStep } from 'hooks/useDeleteCertificateStep';
 import useCertificateStepper from 'hooks/useCertificateStepper';
-import { Trans } from '@lingui/react';
-
+// import { useNavigate } from 'react-router-dom';
+// import useCertificateStepper from 'hooks/useCertificateStepper';
+import { Trans } from '@lingui/macro';
+import { StepperType } from 'types/type';
+import { getStepNumber } from 'components/BasicDetailsForm/util';
+import { StepEnum } from 'types/enums';
 const ConfirmationResetForm = (props: any) => {
-  const navigate = useNavigate();
+  const { step } = props;
+  const { deleteCertificateStep, isDeletingCertificateStep, wasCertificateStepDeleted } =
+    useDeleteCertificateStep(step);
+  const { updateStepStatusState } = useCertificateStepper();
+  // const navigate = useNavigate();
   const { onClose: onAlertClose } = useDisclosure();
-  const { resetForm } = useCertificateStepper();
-  const [isLoading, setIsLoading] = useState<boolean>(false);
+  // const { resetForm } = useCertificateStepper();
+
   const handleOnClose = () => {
     props.onClose();
     onAlertClose();
-    props.onChangeState(false);
   };
+
+  const isResetAllType = props.step === StepEnum.ALL;
+
   const handleResetBtn = () => {
-    setIsLoading(true);
-    // props.onReset(loadDefaultValueFromLocalStorage);
-    resetForm();
-    props.onReset(getRegistrationDefaultValues());
-    props.onChangeResetState(true);
-    props.onChangeState(false);
-    setIsLoading(false);
-    props.onClose();
-    onAlertClose();
-    // props.onRefreshState();
-    navigate('/dashboard/certificate/registration');
+    deleteCertificateStep({
+      key: props.step as StepperType
+    });
+
+    // resetForm();
+    // props.onChangeResetState(true);
+    // props.onChangeState(false);
+
+    // navigate('/dashboard/certificate/registration');
   };
+
+  useEffect(() => {
+    if (wasCertificateStepDeleted) {
+      updateStepStatusState({
+        step: getStepNumber(props.step),
+        status: 'incomplete'
+      });
+      props.onClose();
+    }
+  }, [wasCertificateStepDeleted, props, updateStepStatusState]);
+
   return (
     <>
       <Flex>
         <Box w="full">
-          <Modal closeOnOverlayClick={false} {...props}>
+          <Modal closeOnOverlayClick={false} {...props} isOpen={props.isOpen}>
             <ModalOverlay />
             <ModalContent width={'100%'}>
               <ModalHeader data-testid="confirmation-modal-header" textAlign={'center'}>
-                <Trans id="Clear & Reset Registration Form">Clear & Reset Registration Form</Trans>
+                {isResetAllType ? (
+                  <Trans>Clear & Reset Registration Form</Trans>
```

Comment:
```suggestion
                  <Trans>Clear & Reset Form</Trans>
```

Reply:
This is what I saw before the suggested changes to match the Figma file: 
<img width="491" alt="Screenshot 2023-05-30 at 1 13 07 PM" src="https://github.com/trisacrypto/directory/assets/94616884/e4c1ef94-5068-4d5c-bbe3-edf05615a361">